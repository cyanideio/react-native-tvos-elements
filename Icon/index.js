import Ionicons from 'react-native-vector-icons/Ionicons';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import Entypo from 'react-native-vector-icons/Entypo';
import EvilIcons from 'react-native-vector-icons/EvilIcons';
import Feather from 'react-native-vector-icons/Feather';
import Foundation from 'react-native-vector-icons/Foundation';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import Octicons from 'react-native-vector-icons/Octicons';
import SimpleLineIcons from 'react-native-vector-icons/SimpleLineIcons';
import Zocial from 'react-native-vector-icons/Zocial';

export default class Icon extends Component<Props> {
    render() {
        const { iconType, iconName, iconAlign, iconStyle } = this.props;
        const iconMap = {
            Ionicons,              
            FontAwesome,
            Entypo,
            EvilIcons,
            Feather,
            Foundation,
            MaterialCommunityIcons,
            MaterialIcons,
            Octicons,
            SimpleLineIcons,
            Zocial,
        }
        if (!iconType) { return null }
        var SpecificIcon = iconMap[iconType];
        return  (
            <SpecificIcon style={[
                tvStyle.BtnIcon,
                absoluteAlign(iconAlign), 
                iconStyle,
            ]} 
            name={iconName} />)
    }
}
