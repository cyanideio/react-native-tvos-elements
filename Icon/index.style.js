import { StyleSheet } from 'react-native';
import { deviceHeight, deviceWidth } from '../style';

export default StyleSheet.create({
    BtnIcon: {
        position: 'absolute',
        fontSize: 35,
        color: 'gray' 
    },
});

