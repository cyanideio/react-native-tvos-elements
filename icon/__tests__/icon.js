import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Icon from '../../icon';

describe('Icon Component', () => {
    it('should render without issues', () => {
        const component = shallow(<Icon 
            iconType='Ionicons' 
            iconName='ios-person' 
        />);
        expect(component.length).toBe(1);
        expect(toJson(component)).toMatchSnapshot();
    });
});
