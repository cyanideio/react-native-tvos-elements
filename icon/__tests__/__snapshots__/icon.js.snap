// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Component should render without issues 1`] = `
<Icon
  allowFontScaling={false}
  name="ios-person"
  size={12}
  style={
    Array [
      Object {
        "color": "gray",
        "fontSize": 35,
        "position": "absolute",
      },
      Object {
        "right": 15,
      },
      undefined,
    ]
  }
/>
`;
